src/java.base/share/classes/sun/security/x509/X509CRLImpl.java

```diff
@@ -280,13 +280,16 @@ public X500Principal getIssuerX500Principal() {
          *   prevCertIssuer if it does not exist
          */
         private X500Principal getCertIssuer(X509CRLEntryImpl entry,
-                                            X500Principal prevCertIssuer) {
+                X500Principal prevCertIssuer) throws CRLException {
 
             CertificateIssuerExtension ciExt =
                     entry.getCertificateIssuerExtension();
             if (ciExt != null) {
                 GeneralNames names = ciExt.getNames();
-                X500Name issuerDN = (X500Name) names.get(0).getName();
+                if (!(names.get(0).getName() instanceof X500Name issuerDN)) {
+                    throw new CRLException("Parsing error: "
+                            + "issuer is not an X.500 DN");
+                }
```

Comment:
I checked RFC 5280 and you can have more than one name in the issuer field of the CertificateIssuerExtension, see https://www.rfc-editor.org/rfc/rfc5280#section-5.3.3

But for this code, we are only interested in the X500Name, as we subsequently use that to associate the CRL entry with its issuer. So instead, what you should do is loop thru the names until we find an `X500Name`, and only throw a `CRLException` if we don't find an X500Name. Let me know if this makes sense.

Reply:
Fixed.

Reply:
Does the test need to be modified to test for more than one name? I could go either way.

Reply:
I know how to create a CertificateIssuerExtension with multiple names but I haven't connected the dots as to how to add it to a CRL for testing. It would be an interesting exercise but probably not worth the effort.